### PR TITLE
[CARBONDATA-1761]do not change status of segment during delete segment by ID to marked for delete if it is In progress 

### PR DESCRIPTION
**Pull request by akashrn5**
do not change status of segment to marked for delete if it is in progress

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [*] Any interfaces changed?
 NA
 - [*] Any backward compatibility impacted?
 NA
 - [*] Document update required?
NA
 - [*] Testing done
        tested in local
       
 - [*] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2067/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/409/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1681/

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java [hunk 1]
```diff
@@ -493,6 +493,19 @@ public static void writeLoadDetailsIntoFile(String dataLoadLocation,
             invalidLoadIds.add(loadId);
             return invalidLoadIds;
           }
+          // if the segment status is in progress then no need to delete that.
+          if (SegmentStatus.INSERT_IN_PROGRESS == loadMetadata.getSegmentStatus()) {
+            LOG.error("Cannot delete the Segment which is In Progress. Segment is " + loadId);
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: Please rephrase the sentence 
`"Cannot delete the segemnt "+ loadId +" which is load in progress" `

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java [hunk 2]
```diff
@@ -493,6 +493,19 @@ public static void writeLoadDetailsIntoFile(String dataLoadLocation,
...
+            invalidLoadIds.add(loadId);
+            return invalidLoadIds;
+          }
+          // if the segment status is overwrite in progress, then no need to delete that.
+          if (SegmentStatus.INSERT_OVERWRITE_IN_PROGRESS == loadMetadata.getSegmentStatus()) {
+            LOG.error("Cannot delete the Segment which is Overwrite In Progress. " +
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: Please rephrase the sentence 
`"Cannot delete the segemnt "+ loadId +" which is load overwrite in progress" `

**Comment by akashrn5** [comment]
@ravipesala handled comments please review

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/435/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/436/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2090/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1706/

**Comment by ravipesala** [comment]
retest this please

**Comment by ravipesala** [comment]
retest sdv please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/458/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1725/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/463/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2107/

**Comment by jackylk** [comment]
Please combine this PR with #1610, it is better to keep all rejecting logic in one PR

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1731/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1739/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/513/

**Comment by jackylk** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1774/

**Pull request closed without merging**